controllers/operator/operandrequest.go

```diff
@@ -0,0 +1,237 @@
+									"configmap": "im-datastore-edb-cm",
+								},
+							},
+						},
+					},
+					"registry":          "common-service",
+					"registryNamespace": instance.Namespace,
+				},
+			},
+		},
+	}
+
+	// Create an Unstructured object from the YAML content
+	unstructuredObj := &unstructured.Unstructured{Object: yamlContent}
+
+	err = r.Client.Create(context.TODO(), unstructuredObj)
+	if err != nil {
+		if errors.IsAlreadyExists(err) {
+			reqLogger.Info("OperandRequest" + ctrlCommon.IMEDBOprName + " already exists")
+		} else {
+			reqLogger.Error(err, "Error creating OperandRequest "+ctrlCommon.IMEDBOprName)
+		}
+		return err
+	}
+	return nil
+}
+
+// IM operator creates CommonService CR, to claim the usage on shared embedded database
+func (r *AuthenticationReconciler) createCommonService(instance *operatorv1alpha1.Authentication) (err error) {
+	reqLogger := log.WithValues("Instance.Namespace", instance.Namespace, "Instance.Name", instance.Name)
+	// Define the YAML content as a map[string]interface{}
+	yamlContent := map[string]interface{}{
+		"kind":       "CommonService",
+		"apiVersion": "operator.ibm.com/v3",
+		"metadata": map[string]interface{}{
+			"name":      ctrlCommon.IMEDBCSName,
+			"namespace": instance.Namespace,
+		},
+		"spec": map[string]interface{}{
+			"sharedDBServices": "IM",
+		},
+	}
+
+	unstructuredObj := &unstructured.Unstructured{Object: yamlContent}
+	err = r.Client.Create(context.TODO(), unstructuredObj)
+	if err != nil {
+		if errors.IsAlreadyExists(err) {
+			// Route already exists from a previous reconcile
+			reqLogger.Info("CommonService CR im-common-service already exists")
+		} else {
+			reqLogger.Error(err, "Error creating commonservice "+ctrlCommon.IMEDBCSName)
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *AuthenticationReconciler) checkOperandRequestExists(instance *operatorv1alpha1.Authentication) (exist bool) {
+	reqLogger := log.WithValues("Instance.Namespace", instance.Namespace, "Instance.Name", instance.Name)
+	reqLogger.Info("Query IM EDB OperandRequest", "OperandRequest.Namespace", instance.Namespace, "OperandRequest.Name", common.IMEDBOprName)
+
+	key := types.NamespacedName{Name: ctrlCommon.IMEDBOprName, Namespace: instance.Namespace}
+
+	gvk := schema.GroupVersionKind{
+		Group:   "operator.ibm.com",
+		Version: "v1alpha1",
+		Kind:    "OperandRequest",
+	}
+
+	unstrCert := &unstructured.Unstructured{}
+	unstrCert.SetGroupVersionKind(gvk)
+
+	err := r.Client.Get(context.TODO(), key, unstrCert)
+
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			reqLogger.Info(ctrlCommon.IMEDBOprName + " OperandRequest Not Found")
+			return false
+		}
+		reqLogger.Info("Failed to get OperandRequest " + ctrlCommon.IMEDBOprName)
+		return false
+	}
+	reqLogger.Info("OperandRequest " + ctrlCommon.IMEDBOprName + " is present")
+
+	return true
+}
+
+func (r *AuthenticationReconciler) checkCommonServiceExists(instance *operatorv1alpha1.Authentication) (exist bool) {
```

Comment:
Given that the code isn't doing anything more than trying to create this CommonService CR if it doesn't already exist (versus actually performing comparisons to see if there is a change that needs to be made by the reconciler), breaking out a separate "check" function can be skipped - the logic should just try to create the resource and, if the resource already exists, api will give us back that error, which we can identify to know that there's no more work to do.